pyface/ui/qt4/tasks/tests/test_dock_pane.py

```diff
@@ -75,3 +76,28 @@ def check_panes_and_exit(app_event):
         self.assertTrue(tool_attributes)
         for attr in tool_attributes:
             self.assertTrue(attr)
+
+    def test_dock_windows_undock(self):
```

Comment:
i'm not sure why this test wasn't failing earlier with the in/not in problem but from the looks of it, this test is manually undocking a dock pane and then checks the dock area of the dock panes in the app.

I'm not sure why by default undocking a dockpane means that it's `left` widget area but that's what we seem to be asserting at the end. Do we need to be asserting this at all? Do we care what this value is?

Reply:
I've manually verified that this doesn't crash, so I think the state is good.

The test is testing the dockpane area because it should not change when it is undocked (so yes, we do care about the value) - this was a deliberate choice as one option was to add an "undocked" dockpane area, but the behaviour (particularly when restoring from saved state) was unclear.